### ISSUE TITLE
Style/HashTransformKeys with to_h with a block is too greedy now

### ISSUE THREAD

**Issue by zverok** [rubocop-hq/rubocop#8630]
This code (and any like this, using `Array#to_h` or `Enumerable#to_h`) is a false positive now:
```ruby
%i[a b c].zip([1, 2, 3]).to_h { |name, val| [name, val**2] }
```
↓
```
C: Style/HashTransformValues: Prefer transform_values over to_h {...}.
%i[a b c].zip([1, 2, 3]).to_h { |name, val| [name, val**2] }
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
(After #8517 probably)

**Comment by marcandre** [comment]
cc/ @eugeneius

**Comment by eugeneius** [comment]
This was already a problem for code using `map { ... }.to_h` before https://github.com/rubocop-hq/rubocop/pull/8517:

```bash
$ rubocop _0.89.0_ --only Style/HashTransformValues --stdin foo.rb <<< "%i[a b c].zip([1, 2, 3]).map { |name, val| [name, val**2] }.to_h"
Inspecting 1 file
C

Offenses:

foo.rb:1:1: C: Style/HashTransformValues: Prefer transform_values over map {...}.to_h.
%i[a b c].zip([1, 2, 3]).map { |name, val| [name, val**2] }.to_h
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

Anyway, I opened https://github.com/rubocop-hq/rubocop/pull/8648 with a fix.
